File: leetcode/JavaScript实现/0035.js
// 自己的代码，就是按题目暴力法
// 第一想法就是暴力，第二想法就是二分，后面把代码补上来
var searchInsert = function(nums, target) {
    let len = nums.length;
    for(let i = 0; i < len; ++i) {
        if(nums[i] === target)
            return i;
        else if(target < nums[i] && i === 0)
            return i;
        else if(target > nums[i - 1] && target < nums[i])
            return i;
        else if(target > nums[i] && i === len -1)
            return len;
    }
};

File: leetcode/C++实现/1-99题/0035.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int v_size=nums.size();
        if(v_size==0)
            return 0;
        for(int i=0;i<v_size;++i)
        {
            if(nums[i]==target)
                return i;
            if( nums[i] <target && nums[i+1]>target  )
            {
                return i+1;
            }
        }
        if(target<nums[0])
            return 0;
        if(target>nums[v_size-1])
            return v_size;
        return 0;
    }
};
